### PR TITLE
Api for preview

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

**Comment by davgur** [comment]
@edoshor  Alex check it, can you  make review?

File: Dockerfile [hunk 1]
```diff
@@ -51,6 +51,7 @@ ENV     UWSGI_HTTP_AUTO_CHUNKED=true
 ENV     UWSGI_HTTP_KEEPALIVE=true
 ENV     UWSGI_LAZY_APPS=false
 ENV     UWSGI_WSGI_ENV_BEHAVIOR=holy
+ENV     UWSGI_BUFFER_SIZE=32768
```
Comment (edoshor) [inline Dockerfile]: What is this and why we need it ?

File: preview-client/.eslintcache [hunk 2]
```diff
@@ -0,0 +1 @@
...rchive-unzip/preview-client/src/index.js":"1","/home/david/PycharmProjects/archive-unzip/preview-client/src/App.js":"2","/home/david/PycharmProjects/archive-unzip/preview-client/src/Layout.js":"3"},{"size":197,"mtime":1608258222385,"results":"4","hashOfConfig":"5"},{"size":973,"mtime":1609513281023,"results":"6","hashOfConfig":"5"},{"size":2934,"mtime":1609670064015,"results":"7","hashOfConfig":"5"},{"filePath":"8","messages":"9","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"10"},"1j0d0vd",{"filePath":"11","messages":"12","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"10"},{"filePath":"13","messages":"14","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"/home/david/PycharmProjects/archive-unzip/preview-client/src/index.js",[],["15","16"],"/home/david/PycharmProjects/archive-unzip/preview-client/src/App.js",[],"/home/david/PycharmProjects/archive-unzip/preview-client/src/Layout.js",[],{"ruleId":"17","replacedBy":"18"},{"ruleId":"19","replacedBy":"20"},"no-native-reassign",["21"],"no-negated-in-lhs",["22"],"no-global-assign","no-unsafe-negation"]
```
Comment (edoshor) [inline preview-client/.eslintcache]: should this file be in git ?

File: preview-client/package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.11.4",
+    "@testing-library/react": "^11.1.0",
+    "kmedia-mdb": "https://github.com/Bnei-Baruch/kmedia-mdb.git",
+    "react": "^17.0.1",
+    "react-dom": "^17.0.1",
+    "react-scripts": "4.0.1",
+    "@material-ui/core": "^4.11.2",
+    "@material-ui/icons": "^4.11.2",
+    "jquery": "^3.5.1"
```
Comment (edoshor) [inline preview-client/package.json]: Really ?! 
It's 2021. Are we still using JQuery ?

File: app/preview/views.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            os.makedirs(dir_path)
+        docx_path = save_to_docx(file, dir_path)
+        return send_html(docx_path)
+
+
+def send_html(docx_path):
+    try:
+        path = process_html_path(docx_path)
+    except RuntimeError as ex:
+        return make_response(ex, 400)
```
Comment (edoshor) [inline app/preview/views.py]: RuntimeError are usually 50x (server error)
Try to distinguish between user errors (40x) and server errors (50x)

File: app/preview/views.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    resp = build_response(file)
+    resp.headers.add("Access-Control-Allow-Origin", "*")
+    return resp
+
+
+def build_response(file):
+    if file is None:
+        return make_response("You must send file", 400)
+    with tempfile.TemporaryDirectory() as dir_path:
+        if not os.path.exists(dir_path):
```
Comment (edoshor) [inline app/preview/views.py]: I think `tempfile.TemporaryDirectory()` creates the directory for you. You don't have to do it yourself

File: app/preview/views.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+from app.doc2html.views import process_html_path
+
+htmlByBLobBlueprint = Blueprint('doc2htmlByBLob', __name__)
+
+
+@htmlByBLobBlueprint.route("/doc2htmlByBLob", methods=["POST"])
+def doc_2_html_from_bLob():
+    file = request.files.get('doc')
+    resp = build_response(file)
+    resp.headers.add("Access-Control-Allow-Origin", "*")
```
Comment (edoshor) [inline app/preview/views.py]: CORS should be somewhere generic. not inside specific module (blueprint). Try to google up some best practice for flask and CORS

**Review by edoshor — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (davgur) [reply Dockerfile]: removed

File: preview-client/.eslintcache (hunk 2 above)
Comment (davgur) [reply preview-client/.eslintcache]: removed

File: preview-client/package.json (hunk 3 above)
Comment (davgur) [reply preview-client/package.json]: removed

File: app/preview/views.py (hunk 4 above)
Comment (davgur) [reply app/preview/views.py]: change to 500

File: app/preview/views.py (hunk 5 above)
Comment (davgur) [reply app/preview/views.py]: removed

File: app/preview/views.py (hunk 6 above)
Comment (davgur) [reply app/preview/views.py]: use flask-cors

**Review by edoshor — APPROVED** [review]

**Pull request merged**